lib/datadog.ts

```diff
@@ -32,11 +33,25 @@ export class Datadog extends cdk.Construct {
     super(scope, id);
     this.scope = scope;
     this.props = props;
+
+    if (this.props.extensionLayerVersion !== undefined) {
+      if (this.props.forwarderARN !== undefined) {
+        throw new Error("`extensionLayerVersion` and `forwarderArn` cannot be set at the same time.");
+      }
+      if (this.props.apiKey === undefined && this.props.apiKMSKey === undefined) {
+        throw new Error("When `extensionLayer` is set, `apiKey` or `apiKMSKey` must also be set.");
+      }
+      if (this.props.site === undefined) {
+        throw new Error("When `extensionLayer` is set, `site` must also be set.");
+      }
+    }
+
```

Comment:
I figured this ^ would be the easiest place to check the props and throw the errors accordingly. https://github.com/DataDog/datadog-cdk-constructs/pull/21#issuecomment-784497365

Reply:
This logic looks good, I would extract it into a separate function (`validateProps()` maybe?) and add tests for the validation logic. Also, do we need to validate that if flushMetricsToLogs is false, then an API key must be set?

Reply:
Actually, I see that you already have tests for the validation (nice!!) but I still think it would be a good idea to factor this out into a smaller function. It would actually be easier to write tests for the smaller validation function. But since you already wrote tests for the overarching function, no need to re-do them. They should still work the same with this logic factored out into a smaller function.